packages/integration-tests/src/find_deployed_resource.ts

```diff
@@ -0,0 +1,45 @@
+import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import {
+  CloudFormationClient,
+  DescribeStackResourcesCommand,
+} from '@aws-sdk/client-cloudformation';
+import { BackendIdentifierConversions } from '@aws-amplify/platform-core';
+
+/**
+ * Returns the physical IDs of the resources of type "resourceType" in the stack defined by "backendId"
+ * Traverses nested stacks as well
+ */
+export const findDeployedResources = async (
+  cfnClient: CloudFormationClient,
+  backendId: BackendIdentifier,
+  resourceType: string
+): Promise<string[]> => {
```

Comment:
this could become
```
export const deployedResourcesLookup = new DeployedResourcesLookup(cfnClient);
```
with usage
```
import deployedResourcesLookup
deployedResourcesLookup.lookup(backendId, resourceType);
```

I.e. we could eliminate that cfn client passing. (or we could just keep cfn client as static private in this file).